library/Icingadb/Common/Macros.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+/* Icinga DB Web | (c) 2021 Icinga GmbH | GPLv2 */
+
+namespace Icinga\Module\Icingadb\Common;
+
+use Icinga\Application\Logger;
+use Icinga\Module\Icingadb\Model\Host;
+use Icinga\Module\Icingadb\Model\Service;
+
+trait Macros
+{
+    /**
+     * Caches already processed custom vars query for the given object
+     *
+     * @var array
+     */
+    protected static $loadedCustomVars = [];
```

Comment:
Am I right that you've added this because I mentioned a *performance goal* above? This shouldn't be required for this. The mutation already runs only once, or is at least supposed to, and this already ensures the variables aren't queried more than once.